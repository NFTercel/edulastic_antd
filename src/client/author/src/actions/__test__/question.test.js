import {
  receiveQuestionByIdAction,
  saveQuestionAction,
  setQuestionDataAction,
  setQuestionAlignmentRowAction,
  setQuestionAlignmentRowStandardsAction,
  setQuestionAlignmentAddRowAction,
  setQuestionAction
} from "../question";

describe("Question", () => {
  it("receive question by id should return an action", () => {
    expect(receiveQuestionByIdAction("5c0fd3ea025fa415ffeb6895")).toMatchSnapshot();
    expect(receiveQuestionByIdAction("5c0fd3ea025fa415ffeb6896")).toMatchSnapshot();
    expect(receiveQuestionByIdAction("5c0fd3ea025fa415ffeb6897")).toMatchSnapshot();
  });

  it("save question should return an action", () => {
    expect(saveQuestionAction()).toMatchSnapshot();
  });

  it("set question data should return an action", () => {
    expect(
      setQuestionDataAction({
        data: {
          type: "multipleChoice",
          stimulus: "Which color has the smallest walvelenght?",
          ui_style: {
            type: "horizontal"
          },
          options: [
            {
              value: 0,
              label: "Red"
            },
            {
              value: 1,
              label: "Violet"
            },
            {
              value: 2,
              label: "<p>Green</p>"
            }
          ],
          validation: {
            scoring_type: "exactMatch",
            valid_response: {
              score: 1,
              value: [1]
            },
            alt_responses: []
          },
          multiple_responses: false,
          smallSize: true
        }
      })
    ).toMatchSnapshot();

    expect(
      setQuestionDataAction({
        data: {
          type: "multipleChoice",
          stimulus: "What is the capital city of London?",
          ui_style: {
            type: "horizontal"
          },
          options: [
            {
              value: 0,
              label: "Moscow"
            },
            {
              value: 1,
              label: "Paris"
            },
            {
              value: 2,
              label: "<p>London</p>"
            }
          ],
          validation: {
            scoring_type: "exactMatch",
            valid_response: {
              score: 2,
              value: [2]
            },
            alt_responses: []
          },
          multiple_responses: false,
          smallSize: true
        }
      })
    ).toMatchSnapshot();
  });

  it("set question alignment row should return an action", () => {
    expect(setQuestionAlignmentRowAction(1, 1)).toMatchSnapshot();
    expect(setQuestionAlignmentRowAction(2, 1)).toMatchSnapshot();
  });

  it("set question alignment row standard should return an action", () => {
    expect(setQuestionAlignmentRowStandardsAction(1, 2)).toMatchSnapshot();
    expect(setQuestionAlignmentRowStandardsAction(2, 1)).toMatchSnapshot();
  });

  it("set question alignment add row should return an action", () => {
    expect(setQuestionAlignmentAddRowAction()).toMatchSnapshot();
  });

  it("set question should return an action", () => {
    expect(
      setQuestionAction({
        type: "multipleChoice",
        stimulus: "Which color has the smallest walvelenght?",
        ui_style: {
          type: "horizontal"
        },
        options: [
          {
            value: 0,
            label: "Red"
          },
          {
            value: 1,
            label: "Violet"
          },
          {
            value: 2,
            label: "<p>Green</p>"
          }
        ],
        validation: {
          scoring_type: "exactMatch",
          valid_response: {
            score: 1,
            value: [1]
          },
          alt_responses: []
        },
        multiple_responses: false,
        smallSize: true
      })
    ).toMatchSnapshot();

    expect(
      setQuestionAction({
        type: "multipleChoice",
        stimulus: "How many countries are there in the world?",
        ui_style: {
          type: "horizontal"
        },
        options: [
          {
            value: 0,
            label: "245"
          },
          {
            value: 1,
            label: "256"
          },
          {
            value: 2,
            label: "<p>247</p>"
          }
        ],
        validation: {
          scoring_type: "exactMatch",
          valid_response: {
            score: 2,
            value: [2]
          },
          alt_responses: []
        },
        multiple_responses: false,
        smallSize: true
      })
    ).toMatchSnapshot();
  });
});
