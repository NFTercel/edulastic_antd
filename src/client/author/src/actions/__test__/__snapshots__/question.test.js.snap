// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question receive question by id should return an action 1`] = `
Object {
  "payload": Object {
    "id": "5c0fd3ea025fa415ffeb6895",
  },
  "type": "[question] receive question request",
}
`;

exports[`Question receive question by id should return an action 2`] = `
Object {
  "payload": Object {
    "id": "5c0fd3ea025fa415ffeb6896",
  },
  "type": "[question] receive question request",
}
`;

exports[`Question receive question by id should return an action 3`] = `
Object {
  "payload": Object {
    "id": "5c0fd3ea025fa415ffeb6897",
  },
  "type": "[question] receive question request",
}
`;

exports[`Question save question should return an action 1`] = `
Object {
  "type": "[question] save question request",
}
`;

exports[`Question set question alignment add row should return an action 1`] = `
Object {
  "payload": Object {
    "alignmentRow": undefined,
  },
  "type": "[question] set question alignment add row",
}
`;

exports[`Question set question alignment row should return an action 1`] = `
Object {
  "payload": Object {
    "alignmentRow": 1,
    "index": 1,
  },
  "type": "[question] set question alignment row",
}
`;

exports[`Question set question alignment row should return an action 2`] = `
Object {
  "payload": Object {
    "alignmentRow": 1,
    "index": 2,
  },
  "type": "[question] set question alignment row",
}
`;

exports[`Question set question alignment row standard should return an action 1`] = `
Object {
  "payload": Object {
    "alignmentStandards": 2,
    "index": 1,
  },
  "type": "[question] set question alignment row standards",
}
`;

exports[`Question set question alignment row standard should return an action 2`] = `
Object {
  "payload": Object {
    "alignmentStandards": 1,
    "index": 2,
  },
  "type": "[question] set question alignment row standards",
}
`;

exports[`Question set question data should return an action 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "data": Object {
        "multiple_responses": false,
        "options": Array [
          Object {
            "label": "Red",
            "value": 0,
          },
          Object {
            "label": "Violet",
            "value": 1,
          },
          Object {
            "label": "<p>Green</p>",
            "value": 2,
          },
        ],
        "smallSize": true,
        "stimulus": "Which color has the smallest walvelenght?",
        "type": "multipleChoice",
        "ui_style": Object {
          "type": "horizontal",
        },
        "validation": Object {
          "alt_responses": Array [],
          "scoring_type": "exactMatch",
          "valid_response": Object {
            "score": 1,
            "value": Array [
              1,
            ],
          },
        },
      },
    },
  },
  "type": "[question] set question data",
}
`;

exports[`Question set question data should return an action 2`] = `
Object {
  "payload": Object {
    "data": Object {
      "data": Object {
        "multiple_responses": false,
        "options": Array [
          Object {
            "label": "Moscow",
            "value": 0,
          },
          Object {
            "label": "Paris",
            "value": 1,
          },
          Object {
            "label": "<p>London</p>",
            "value": 2,
          },
        ],
        "smallSize": true,
        "stimulus": "What is the capital city of London?",
        "type": "multipleChoice",
        "ui_style": Object {
          "type": "horizontal",
        },
        "validation": Object {
          "alt_responses": Array [],
          "scoring_type": "exactMatch",
          "valid_response": Object {
            "score": 2,
            "value": Array [
              2,
            ],
          },
        },
      },
    },
  },
  "type": "[question] set question data",
}
`;

exports[`Question set question should return an action 1`] = `
Object {
  "payload": Object {
    "data": Object {
      "multiple_responses": false,
      "options": Array [
        Object {
          "label": "Red",
          "value": 0,
        },
        Object {
          "label": "Violet",
          "value": 1,
        },
        Object {
          "label": "<p>Green</p>",
          "value": 2,
        },
      ],
      "smallSize": true,
      "stimulus": "Which color has the smallest walvelenght?",
      "type": "multipleChoice",
      "ui_style": Object {
        "type": "horizontal",
      },
      "validation": Object {
        "alt_responses": Array [],
        "scoring_type": "exactMatch",
        "valid_response": Object {
          "score": 1,
          "value": Array [
            1,
          ],
        },
      },
    },
  },
  "type": "[question] set question",
}
`;

exports[`Question set question should return an action 2`] = `
Object {
  "payload": Object {
    "data": Object {
      "multiple_responses": false,
      "options": Array [
        Object {
          "label": "245",
          "value": 0,
        },
        Object {
          "label": "256",
          "value": 1,
        },
        Object {
          "label": "<p>247</p>",
          "value": 2,
        },
      ],
      "smallSize": true,
      "stimulus": "How many countries are there in the world?",
      "type": "multipleChoice",
      "ui_style": Object {
        "type": "horizontal",
      },
      "validation": Object {
        "alt_responses": Array [],
        "scoring_type": "exactMatch",
        "valid_response": Object {
          "score": 2,
          "value": Array [
            2,
          ],
        },
      },
    },
  },
  "type": "[question] set question",
}
`;
